### PR TITLE
Explore: Fix always disabled QueryField for InfluxDB

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
When InfluxDB has been selected as a datasource for Logs in Explore, QueryField button with Measurements has been disabled, even though, InfluxDB had measurements.  This was due to the fact, that `disabled={!measurement}` [was used](https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/influxdb/components/InfluxLogsQueryField.tsx#L146) to determine this. However, we setState of measurement only in `onMeasurementsChange` that is passed to Cascader and its value is changed only if user selects one of the values. This is not possible, as QueryField button is disabled. 

This PR fixes this by determining, if QueryField button based on the fact, if measurments are present.

### CODE REVIEW

File: public/app/plugins/datasource/influxdb/components/InfluxLogsQueryField.tsx [hunk 1]
```diff
@@ -133,6 +133,7 @@ export class InfluxLogsQueryField extends React.PureComponent<Props, State> {
     const { datasource } = this.props;
     const { measurements, measurement, field, error } = this.state;
     const cascadeText = getChooserText({ measurement, field, error });
+    const hasMeasurement = measurements && measurements.length > 0;
```
Comment (hugohaggmark) [inline public/app/plugins/datasource/influxdb/components/InfluxLogsQueryField.tsx]: Can we extract this to a memoized function? That way you can probably also add simple tests.

**Review by hugohaggmark — APPROVED** [review]
LGTM! Left a small comment that you can look into if it makes sense.

File: public/app/plugins/datasource/influxdb/components/InfluxLogsQueryField.tsx (hunk 1 above)
Comment (ivanahuckova) [reply public/app/plugins/datasource/influxdb/components/InfluxLogsQueryField.tsx]: Absolutely! Good point.

Comment (ivanahuckova) [reply public/app/plugins/datasource/influxdb/components/InfluxLogsQueryField.tsx]: @hugohaggmark I have talked about memoizing `hasMeasurments` with @aocenas and also, I implemented and tested it. Based on both, it probably doesn't make sense to use memoization. Couple reasons: 
1. After implementing it, there weren't any significant performance improvements.
2. Code was simpler without memoization, no import necessary.
3. As there is no reason for `hasMeasurments` to be global function, I am not sure what kind of tests should be implemented, as memoize-one has implemented tests for its memoization.
Therefore, as there aren't any significant performance gains, there is probably no major reasons to memoize `hasMeasurments`.  

I am going to merge this PR, as this should be fixed in 6.5,  but let me know, if there is any reason why you think we should still implement this.

**Review by davkal — APPROVED** [review]

**Pull request merged**